java/flight/src/main/java/org/apache/arrow/flight/CallStatus.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.util.Objects;
+
+/**
+ * The result of a Flight RPC.
```

Comment:
can you expand on this comment.  In particular, it is not immediately obvious why this can't just be an exception class (i.e. why do we need to represent OK)?  It seems like this might have do with the flight state machine if I understand the code correctly.

Reply:
I've mostly followed the gRPC API in this regard, which has both checked and unchecked versions of the exception that use a common status object. We don't need to represent OK and don't use the status, so I've removed it. I did expand the comment a bit.